KH2: Region Rework and Optimization

JaredWeakStrike opened the pull request:
## What is this fixing or adding?
breaking up the logic into its own functions for each difficulty setting. Previously, it ran all difficulty logic and just returned the true/false for the option in the yaml instead of just running the single difficulty

Also added some logic for some prevalent roadblocks that didnt have any logic before
## How was this tested?
Ran unit tests for each of the settings and it generated fine 

## If this makes graphical changes, please attach screenshots.

JaredWeakStrike committed:
adding qwints suggestions

JaredWeakStrike committed:
add stat increase protection and ingame yml stuff

JaredWeakStrike committed:
idk how I forgot these

JaredWeakStrike committed:
reword things

JaredWeakStrike committed:
Update worlds/kh2/Client.py

Co-authored-by: qwint <qwint.42@gmail.com>

JaredWeakStrike committed:
3.12 compat

JaredWeakStrike committed:
Merge branch 'kh2-optimize-client' of https://github.com/JaredWeakStrike/Archipelago into kh2-optimize-client

JaredWeakStrike committed:
too long of a line

JaredWeakStrike committed:
why didnt I do this before lol

JaredWeakStrike committed:
reading is hard

JaredWeakStrike committed:
missed one

JaredWeakStrike committed:
forgot the self

JaredWeakStrike committed:
fix crash if you get datapackage that isnt kh2

JaredWeakStrike committed:
ripping stuff from puzzle pieces branch

JaredWeakStrike committed:
update to main?

JaredWeakStrike committed:
Merge remote-tracking branch 'upstream/main' into kh2-optimize-client

JaredWeakStrike committed:
update to use 0.10 as base and fix violet's base 0 on hex values

JaredWeakStrike committed:
reverting this because I'm bad at my job

JaredWeakStrike committed:
fix the dict and get it to run

also restored indirect condition for final form

JaredWeakStrike committed:
add location rules and clean up some logic

JaredWeakStrike committed:
add corridors logic and clean up summon levels

JaredWeakStrike committed:
kh ascii and display async func

JaredWeakStrike committed:
first implementation of popups

JaredWeakStrike committed:
cmd commands and recieve logic

JaredWeakStrike committed:
everything works now just to save it to a json

JaredWeakStrike committed:
make client settings save to a json

JaredWeakStrike committed:
Merge branch 'ArchipelagoMW:main' into kh2-displayitems-ingame

JaredWeakStrike committed:
check if the infobar is active before setting

check if the infobar is active before setting

JaredWeakStrike committed:
Fix f string syntax error

JaredWeakStrike committed:
deathlink

JaredWeakStrike committed:
break the client into multiple files

JaredWeakStrike committed:
.

JaredWeakStrike committed:
rip out recieve handeling

JaredWeakStrike committed:
amount_of_items overflow

JaredWeakStrike committed:
Merge remote-tracking branch 'upstream/main' into kh2-displayitems-ingame

JaredWeakStrike committed:
comments

JaredWeakStrike committed:
move file and deathlink text

JaredWeakStrike committed:
update message with period

JaredWeakStrike committed:
backwards compat will revert

JaredWeakStrike committed:
a

JaredWeakStrike committed:
party member dicts

JaredWeakStrike committed:
equipment now supports multiple and no dupes

JaredWeakStrike committed:
fix missable death links and launcher import

nicholassaylor reviewed (approved):
PR is mostly helper functions and variable renames, making the PR seem bigger than it might actually be. After going through everything and running some generations, it all LGTM.

JaredWeakStrike committed:
Merge remote-tracking branch 'upstream/main' into deathlink

JaredWeakStrike committed:
Merge remote-tracking branch 'upstream/main' into deathlink

JaredWeakStrike committed:
merge the cache locations warning

JaredWeakStrike committed:
fix client crash on invalid file name and fail generation if bad yaml name

JaredWeakStrike committed:
fix client crashing if json is broke

JaredWeakStrike committed:
Merge branch 'ArchipelagoMW:main' into deathlink

JaredWeakStrike committed:
add init file to pass unit test

JaredWeakStrike committed:
updated item lookups to not be the deprecated versions

JaredWeakStrike committed:
Merge remote-tracking branch 'upstream/main' into deathlink

JaredWeakStrike committed:
Merge branch 'ArchipelagoMW:main' into deathlink

JaredWeakStrike committed:
update docs for deathlink mod and such

JaredWeakStrike committed:
fix in game item display and softlock when inputting the wrong name

lallaria committed:
Adding AP-style KH2 icons

JaredWeakStrike committed:
no more wqerkzeug and fixed recieve popup changing

JaredWeakStrike committed:
Merge branch 'ArchipelagoMW:main' into deathlink

JaredWeakStrike committed:
Merge remote-tracking branch 'upstream/main' into deathlink

JaredWeakStrike committed:
Merge branch 'deathlink' of https://github.com/JaredWeakStrike/Archipelago into deathlink

JaredWeakStrike committed:
updating some docs

JaredWeakStrike committed:
clean this up to not spread lies

JaredWeakStrike committed:
magic no longer on pause

JaredWeakStrike committed:
first commit for chest stuff

JaredWeakStrike committed:
steam support

JaredWeakStrike committed:
min of 4 for drive

JaredWeakStrike committed:
bug cleanup

clean up some duplicate code and cleanup some user stuff like input.lower to make it not as case sensitive

JaredWeakStrike committed:
Merge branch 'ArchipelagoMW:main' into deathlink

JaredWeakStrike committed:
Merge branch 'deathlink' into kh2icons

JaredWeakStrike committed:
Merge pull request #10 from lallaria/kh2icons

Add AP-style KH2 icons to client and OpenKH zip

JaredWeakStrike committed:
fixed some things with the img and seed

JaredWeakStrike committed:
Merge branch 'ArchipelagoMW:main' into kh2-regeion-rework

JaredWeakStrike committed:
logic overrides for fight logic

JaredWeakStrike committed:
clean it up

JaredWeakStrike committed:
fix some things missing and adding pr minutie fight loigic

duckboycool commented:
This does have conflicts currently. I'm not sure if the changes here would be the same after and if it could be reviewed as is though.

JaredWeakStrike committed:
Merge remote-tracking branch 'upstream/main' into kh2-regeion-rework

duckboycool reviewed (approved):
I didn't look too closely at the rules changes and also didn't test in game, but the code changes here seem fine and it got a similar error rate compared to currently with the fuzzer.
